lib/cn/kvs_mblk_desc.h

```diff
@@ -3,23 +3,70 @@
  * Copyright (C) 2015-2022 Micron Technology, Inc.  All rights reserved.
  */
 
-#ifndef HSE_KVS_MBLK_DESC_H
-#define HSE_KVS_MBLK_DESC_H
+#ifndef HSE_MBLK_DESC_H
+#define HSE_MBLK_DESC_H
 
 #include <stdint.h>
 
+#include <hse/error/merr.h>
+
+#include <hse_util/page.h>
+
 #include <hse/types.h>
 
-struct mpool_mcache_map;
+/* MTF_MOCK_DECL(mblk_desc) */
+
 struct mpool;
 
+/* Mblock descriptor */
 struct kvs_mblk_desc {
```

Comment:
Not really related to this PR, but why does this struct not live in mpool?

```c
merr_t
mpool_mblock_get_descriptor(struct mpool *mp, uint64_t mbid, struct mpool_mblock_desc *desc);
```

Reply:
cc @nabeelmmd

Reply:
It's a descriptor for what KVDB considers an "open" mblock (i.e., "memory mapped and part of a kvset"). Mpool doesn't have the concept of an open or closed mblock -- all mpool mblock APIs use the mblock ID.  It also caches properties key to KVDB in a condensed form,